### ISSUE TITLE
How AO and VO model are trained?

### ISSUE THREAD

**Issue by omerfdeniz** [lordmartian/deep_avsr#54]
Thanks for sharing such a useful repo.

You mention that:

"We train the AO and VO models first. We then initialize the AV model with weights from the trained AO and VO models as follows: AO Audio Encoder → AV Audio Encoder, VO Video Encoder → AV Video Encoder, VO Video Decoder → AV Joint Decoder."

You use AO and VO to initialize AV for further pre-training on LRS2 pre-train set and fine-tuning on LRS-2 train set. But when you pre-train AO and VO before that initialization, do you only use the pre-train set of LRS-2 or do you pre-train on pre-train+train sets or do you pre-train on pre-train and fine-tune on train set before using AO and VO for AV model training? This is particularly important for us as we will use AO and VO model weights you shared and want to know which LRS2 sets they have seen during training.

Thanks!

**Comment by smeetrs** [comment]
The AO and VO models are initialized with random weights, then pretrained on the pretrain set and then fine-tuned on the train set. Once training for both models is complete, they are used to initialize the AV model.

**Comment by liuxiaozhen123** [comment]
Hello, after training separate audio and video models in the "audio_only" and "video_only" folders, I'm trying to use the weights from the audio_only model to initialize the AV model as pre-trained weights. However, I'm experiencing some loss of weights. How can I implement the initialization of the AV Audio Encoder using the AO Audio Encoder?
